types/make-fetch-happen/make-fetch-happen-tests.ts

```diff
@@ -0,0 +1,74 @@
+import fetcher from 'make-fetch-happen';
+import { Integrity } from 'ssri';
+import { URL as NodeURL } from 'url';
+
+// Needs arguments when invoked
+// $ExpectError
+fetcher();
+
+// Works with no defaults applied.
+// $ExpectType FetchInterface
+fetcher.defaults();
+
+// When used recursively, still needs arguments when invoked.
+// $ExpectError
+fetcher.defaults()();
+
+// $ExpectType FetchInterface
+fetcher.defaults();
+
+// Returns a Response when fetched.
+// $ExpectType Promise<Response>
+fetcher('http://url');
+
+// $ExpectType Promise<Response>
+fetcher.defaults('http://default-uri')({
+    cert: 'MY_PEM',
+});
+
+// $ExpectType Promise<Response>
+fetcher('http://default-uri', {
+    cert: 'MY_PEM',
+});
+
+// $ExpectType Promise<Response>
+fetcher.defaults()('http://url');
+
+// $ExpectType Promise<Response>
+fetcher.defaults().defaults()('http://url');
+
+// $ExpectError
+fetcher('https://secure', { cache: "invalid-option" });
```

Comment:
Would it be useful to also test a correct cache option as well? I can form a counterargument against since the `RequestCache` type is external and potentially a moving target, though an option like `default` is unlikely to change or disappear from that union of strings.